test/build-images.sh

```diff
@@ -40,20 +32,19 @@ then
   echo "docker images for api-server, frontend, scheduledworkflow and \
     persistenceagent are already built in ${GCR_IMAGE_BASE_DIR}."
 else
-  echo "submitting argo workflow to build docker images for commit ${PULL_PULL_SHA}..."
-  # Build Images
-  ARGO_WORKFLOW=`argo submit ${DIR}/build_image.yaml \
-  -p image-build-context-gcs-uri="$remote_code_archive_uri" \
-  ${IMAGE_BUILDER_ARG} \
-  -p api-image="${GCR_IMAGE_BASE_DIR}/api-server" \
-  -p frontend-image="${GCR_IMAGE_BASE_DIR}/frontend" \
-  -p scheduledworkflow-image="${GCR_IMAGE_BASE_DIR}/scheduledworkflow" \
-  -p persistenceagent-image="${GCR_IMAGE_BASE_DIR}/persistenceagent" \
-  -n ${NAMESPACE} \
-  --serviceaccount test-runner \
-  -o name
-  `
-  echo "build docker images workflow submitted successfully"
-  source "${DIR}/check-argo-status.sh"
-  echo "build docker images workflow completed"
+  echo "submitting cloud build to build docker images for commit ${PULL_PULL_SHA}..."
+  IMAGES_BUILDING=true
+  CLOUD_BUILD_COMMON_ARGS=(. --async --format='value(id)' --substitutions=_GCR_BASE=${GCR_IMAGE_BASE_DIR})
+  # Split into two tasks because api_server builds slowly, use a separate task
+  # to make it faster.
+  BUILD_ID_API_SERVER=$(gcloud builds submit ${CLOUD_BUILD_COMMON_ARGS[@]} \
```

Comment:
Is the gcloud build wraps the current directory for the source by default?

Reply:
Yes, it wraps current directory. I also tried to find if there's an option that can let us reuse existing gcs object. However, I didn't find such an option.